src/data-source/collection-browser-data-source.ts

```diff
@@ -246,6 +246,19 @@ export class CollectionBrowserDataSource
     this.requestHostUpdate();
   }
 
+  /**
+   * @inheritdoc
+   */
+  resetPages(): void {
+    if (Object.keys(this.pages).length !== this.host.maxPagesToManage) {
+      this.pages = {};
+    }
+
+    // Invalidate any fetches in progress
+    this.fetchesInProgress.clear();
+    this.requestHostUpdate();
```

Comment:
Q: Should these lines be inside the `if` block too, or do we need them to run even if we don't reset?

Reply:
We call the resetPages when needed. so inside/outside both work. I can just moving these inside. Thanks!